cms/tests/test_reversion_tests.py

```diff
@@ -231,6 +232,61 @@ def test_recover(self):
             # test that CMSPlugin subclasses are recovered
             self.assertEqual(Text.objects.all().count(), 1)
 
+    def test_recover_with_apphook(self):
+        """
+        Test that you can recover a page
+        """
+        from cms.utils.helpers import make_revision_with_plugins
+        from cms.utils.reversion_hacks import create_revision, revision_manager
+
+        self.assertEqual(Page.objects.count(), 2)
+
+        with self.login_user_context(self.user):
+
+            page_data = self.get_new_page_data_dbfields()
+            page_data['apphook'] = 'SampleApp'
+            page_data['apphook_namespace'] = 'SampleApp'
+
+            with create_revision():
+                # Need to create page manually to add apphooks
```

Comment:
I am using API below in ``page = create_page(**page_data)``.
that's what I meant by "manually".